File: frontend/src/services/AttendeeService.js
import axios from 'axios';

const ATTENDEE_API_BASE_URL = "http://localhost:8080/api/attendees";

class AttendeeService {
    getAllAttendees() {
        return axios.get(ATTENDEE_API_BASE_URL);
    }

    getAttendeeById(id) {
        return axios.get(`${ATTENDEE_API_BASE_URL}/${id}`);
    }

    createAttendee(attendee) {
        return axios.post(ATTENDEE_API_BASE_URL, attendee);
    }

    updateAttendee(id, attendee) {
        return axios.put(`${ATTENDEE_API_BASE_URL}/${id}`, attendee);
    }

    deleteAttendee(id) {
        return axios.delete(`${ATTENDEE_API_BASE_URL}/${id}`);
    }
}

export default new AttendeeService();

File: src/main/java/Airlines/AirlineTicket/Attendee/Models/AttendeeService.java
package Airlines.AirlineTicket.Attendee.Models;
import Airlines.AirlineTicket.Attendee.Attendee;

import java.util.List;


public interface AttendeeService {


    public List<Attendee> getAllAttendees();

    public Attendee getAttendeeById(Long id);

    public Attendee saveAttendee(Attendee attendee);

    public void deleteAttendee(Long id);

    }